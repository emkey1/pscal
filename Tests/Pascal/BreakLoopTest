program BreakLoopTest;
uses CRT; // Include CRT for WriteLn

var
  i: integer;
  ch: char;

begin
  WriteLn('Testing FOR loop break...');
  for i := 1 to 10 do
  begin
    Write(i:3);
    if i = 5 then
    begin
      WriteLn; // Newline before break message
      WriteLn('Breaking FOR loop at i = ', i);
      Break;
    end;
  end;
  WriteLn('After FOR loop.');
  WriteLn;

  WriteLn('Testing WHILE loop break...');
  i := 0;
  while i < 10 do
  begin
    i := i + 1;
    Write(i:3);
    if i = 6 then
    begin
      WriteLn; // Newline before break message
      WriteLn('Breaking WHILE loop at i = ', i);
      Break;
    end;
  end;
  WriteLn('After WHILE loop.');
  WriteLn;

  WriteLn('Testing REPEAT loop break...');
  i := 0;
  repeat
    i := i + 1;
    Write(i:3);
    if i = 7 then
    begin
       WriteLn; // Newline before break message
       WriteLn('Breaking REPEAT loop at i = ', i);
       Break;
    end;
  until i >= 10; // Condition would normally stop it at 10
  WriteLn('After REPEAT loop.');
  WriteLn;

end.
