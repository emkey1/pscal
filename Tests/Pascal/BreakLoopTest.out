Testing FOR loop break...
  1  2  3  4  5
Breaking FOR loop at i = 5
After FOR loop.

Testing WHILE loop break...
  1  2  3  4  5  6
Breaking WHILE loop at i = 6
After WHILE loop.

Testing REPEAT loop break...
  1  2  3  4  5  6  7
Breaking REPEAT loop at i = 7
After REPEAT loop.

